lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -454,48 +460,86 @@ def get_dependencies(self, candidate):
             return []
 
         return [
-            self._make_req_from_dict({'name': dep_name, 'version': dep_req})
+            self._make_req_from_dict(  # type: ignore[misc]
+                {
+                    'name': dep_name,
+                    'version': dep_req,
+                }
+            )
             for dep_name, dep_req in req_map.items()
         ]
 
 
 # Classes to handle resolvelib API changes between minor versions for 0.X
 class CollectionDependencyProvider050(CollectionDependencyProviderBase):
-    def find_matches(self, requirements):  # type: ignore[override]
-        # type: (list[Requirement]) -> list[Candidate]
+    def find_matches(
+            self,
+            requirements: list[Requirement],
+    ) -> list[Candidate]:  # type: ignore[override]
```

Comment:
Since we're cleaning up the type annotations, we should be able to eliminate the ignores.